### PR TITLE
Improved callback handler - RPC

### PR DESCRIPTION
**Pull request by momintlh**
Swapped RPC register callbacks with `CallBackManager`

### CODE REVIEW

File: Assets/PlayroomKit/modules/CallbackManager.cs [hunk 1]
```diff
@@ -27,14 +27,25 @@ public static string RegisterCallback(Delegate callback, string key = null)
...
-                else Debug.LogError($"Callback with key {key} is of unsupported type!");
+                switch (callback)
+                {
+                    case Action action:
+                        action?.Invoke();
+                        break;
+                    case Action<string> stringAction:
+                        stringAction?.Invoke(arg1);
+                        break;
+                    case Action<string, string> doubleStringAction:
```
Comment (SaadBazaz) [inline Assets/PlayroomKit/modules/CallbackManager.cs:42]: Will we have to add another switch statement for every type of argument?

Comment (momintlh) [reply Assets/PlayroomKit/modules/CallbackManager.cs:42]: Yes, <string, string> is the max we are using in the SDK so far.

Comment (SaadBazaz) [reply Assets/PlayroomKit/modules/CallbackManager.cs]: Does this impact user-provided callbacks? Can the user add any callback which they want?

Comment (momintlh) [reply Assets/PlayroomKit/modules/CallbackManager.cs]: Doesn't impact afaik, data is in string for example RPC sends : data and senderId

Comment (SaadBazaz) [reply Assets/PlayroomKit/modules/CallbackManager.cs]: Can't the data be in number or object format? What if the user sends that? Can we serialize/deserialize it?

**Review by SaadBazaz — APPROVED** [review]
If looks alright after resolving the comment, go ahead for merge.
Can improve as we go. Just ensure this is **fully tested**.

**Pull request merged**